packages/plugin-ext/src/main/browser/authentication-main.ts

```diff
@@ -197,10 +198,12 @@ export class AuthenticationMainImpl implements AuthenticationMain {
         return allow;
     }
 
-    protected async loginPrompt(providerName: string, extensionName: string, recreatingSession: boolean, _detail?: string): Promise<boolean> {
-        const message = recreatingSession
+    protected async loginPrompt(providerName: string, extensionName: string, recreatingSession: boolean, detail?: string): Promise<boolean> {
+        let message = recreatingSession
             ? nls.localizeByDefault("The extension '{0}' wants you to sign in again using {1}.", extensionName, providerName)
             : nls.localizeByDefault("The extension '{0}' wants to sign in using {1}.", extensionName, providerName);
+
+        message = detail ? message + ' ' + detail : message;
```

Comment:
A couple of questions here:

1. Shouldn't we include the detail as a parameter of the internationalized message?
2. What kind of text is typically here: a whole paragraph, a single sentence?
3. If this text can be longer, shouldn't it be separated from the main text at least with a line break?

Reply:
I just checked in VS Code: they use a modal dialog and not an info dialog. Using a proper dialog would also allow to format as we like.

Reply:
I used a modal dialog rather than a notification. I still have a small issue on the "Allow" button internationalization of this dialog. 
In the current version, I do not run any internationalization on "Allow" button.

Reply:
![modal-dialog](https://github.com/eclipse-theia/theia/assets/3964263/a1711cdc-67ca-42fd-b810-665fe5643e8d)

Reply:
@rschnekenbu what's the problem with localization? Maybe @msujew  can help.

Reply:
While developing this PR, I stumbled onto an issue with localization. vscode uses the following code for the "Allow" button: 
`primaryButton: nls.localize({ key: 'allow', comment: ['&& denotes a mnemonic'] }, "&&Allow")`

I have the same kind of code on the dialog, using nls.localizeByDefault("&&Allow"). As I do not have any locale on my debug environement, I get "&&Allow" as the text of the button. 
@msujew, would you have an idea  how I could get the correct text? 
https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/nls.ts#L35 => this part of the code is skipped, as localization is undefined. So we don't normalize the string.

Reply:
Theia strips any mnemonics off of vscode's original text. You should be able to just use `nls.localizeByDefault('Allow')` (assuming it's in the `nls.metadata.json` file).

Reply:
Thanks, Mark! Removing the mnemonics worked.

Reply:
![login-english](https://github.com/eclipse-theia/theia/assets/3964263/6db7523d-1ef8-4d16-add6-6986d3202dc2)
Login in English
![login-french](https://github.com/eclipse-theia/theia/assets/3964263/20568527-dd9e-4f43-9122-8a2233c92df6)
Login in French